processor/src/main/java/com/epam/learn/processor/service/SongProcessorService.java

```diff
@@ -0,0 +1,61 @@
+package com.epam.learn.processor.service;
+
+import com.epam.learn.processor.dto.SongDTO;
+import lombok.SneakyThrows;
+import org.apache.tika.exception.TikaException;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.parser.ParseContext;
+import org.apache.tika.parser.Parser;
+import org.apache.tika.parser.mp3.Mp3Parser;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import org.xml.sax.ContentHandler;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+@Service
+public class SongProcessorService implements ProcessorService {
+
+    private static final String NAME_KEY = "dc:title";
+    private static final String ARTIST_KEY = "xmpDM:artist";
+    private static final String ALBUM_KEY = "xmpDM:album";
+    private static final String LENGTH_KEY = "xmpDM:duration";
+    private static final String YEAR_KEY = "xmpDM:releaseDate";
+
+    @SneakyThrows
```

Comment:
That's a 'dangerous' annotation. Especially when you are working with resources that might lead to leakage. It is better to handle the exceptions explicitly.

Reply:
Replaced